src/error.rs

```diff
@@ -0,0 +1,83 @@
+use core::fmt;
```

Comment:
Oh, sorry, I just noticed this: You didn't add the license header to the top of the file. Please do so and I'll merge ASAP.